inventory/sample/group_vars/all/offline.yml

```diff
@@ -32,8 +32,7 @@
 # etcd_download_url: "{{ files_repo }}/github.com/etcd-io/etcd/releases/download/{{ etcd_version }}/etcd-{{ etcd_version }}-linux-{{ image_arch }}.tar.gz"
 
 # [Optional] Calico: If using Calico network plugin
-# calicoctl_download_url: "{{ files_repo }}/github.com/projectcalico/calico/releases/download/{{ calico_ctl_version }}/calicoctl-linux-{{ image_arch }}"
-# calicoctl_alternate_download_url: "{{ files_repo }}/github.com/projectcalico/calicoctl/releases/download/{{ calico_ctl_version }}/calicoctl-linux-{{ image_arch }}"
+# calicoctl_download_url: "{{ files_repo }}/github.com/projectcalico/{{ 'calico' if (calico_ctl_version is version('v3.21.3','>=')) else 'calicoctl' }}/releases/download/{{ calico_ctl_version }}/calicoctl-linux-{{ image_arch }}"
```

Comment:
3.22.0 maybe better than the 3.21.3.

Every version <=3.21.x can be found at the  https://github.com/projectcalico/calicoctl
Every version >=3.22.x can be found at the  https://github.com/projectcalico/calico

Reply:
Honestly until we drop calico < 3.21 I would keep this because it was introduced specifically with calico breaking release artefacts mid-cycle. The errors you are seeing are aesthetic only and allow us to deploy older versions of calico.

/hold

Reply:
A similar solution (version filter) is already used in several places: [one](https://github.com/kubernetes-sigs/kubespray/blob/master/roles/download/defaults/main.yml#L972), [two](https://github.com/kubernetes-sigs/kubespray/blob/master/roles/download/defaults/main.yml#L1514), [three](https://github.com/kubernetes-sigs/kubespray/blob/master/roles/download/defaults/main.yml#L1516).

Reply:
@oomichi WDYT ?

Reply:
Sorry for my late comment.

I prefer current simple lines without a condition in the middle of the line.
In addition, calico 3.21 will be un-maintained soon according to https://github.com/projectcalico/calico/branches
So we might drop calico 3.21 simply by removing `calicoctl_alternate_download_url`

Reply:
Closing then, are you align with this @JiffsMaverick ?

Reply:
> I'm just very interested in making sure there are no errors while the playbooks are running.

Yeah, I see that point.
When I faced some issue at Kubespray ansible-playbook run, I investigated the output errors.
And if the unrelated errors were output, that made the investigation difficult and mess.
It is very nice to reduce such unrelated errors as possible, I completely agree with you.

> Therefore, I want to know the date (or a way to understand that support is over) so that I can send a new pull request to fix these errors as soon as possible.

OK, Calico v3.21 branch is not updated at all since 2 months ago.
And the version is our lowest supported version since https://github.com/kubernetes-sigs/kubespray/pull/8935
So let me propose a pull request to drop the version.

BTW an issue https://github.com/kubernetes-sigs/kubespray/issues/9514 is reported.
The other people also would be facing the same issue.